### PR TITLE
[docs] add Development Client initial reference page

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-10836

# How

An initial version of the Development Client reference page.

# Test Plan

The changes have been reviewed locally.

# Preview

screencapture-localhost-3002-versions-latest-sdk-dev-client-2024-01-25-17_28_39

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/dev-client.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import APISection from '~/components/plugins/APISection';
+import { APIInstallSection } from '~/components/plugins/InstallSection'
+import {
+  ConfigPluginExample,
+  ConfigPluginProperties,
+} from '~/components/plugins/ConfigSection';
+import PlatformsSection from '~/components/plugins/PlatformsSection';
+
+`expo-dev-client` is a library which can be installed in any Expo and React Native project to create a [development build](https://docs.expo.dev/workflow/overview/#development-builds).
```
Comment (Simek) [inline docs/pages/versions/unversioned/sdk/dev-client.mdx:17]:
```suggestion
`expo-dev-client` is a library which can be installed in any Expo and React Native project to create a [development build](/workflow/overview/#development-builds).
```

Copy pasta issue.

File: docs/pages/versions/unversioned/sdk/dev-client.mdx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  ConfigPluginExample,
+  ConfigPluginProperties,
+} from '~/components/plugins/ConfigSection';
+import PlatformsSection from '~/components/plugins/PlatformsSection';
+
+`expo-dev-client` is a library which can be installed in any Expo and React Native project to create a [development build](/workflow/overview/#development-builds).
+
+A development build is a debug build of your app that provides a flexible, reliable and a complete development environment. `expo-dev-client` provides an extensible in-app development tools UI, launcher UI for that debug build and ability to load project using Expo CLI. These debug builds are also shareable with your team or on multiple devices when working or reviewing that app is in development.
+
+After adding this library, the app's release builds are not affected other than a few header files.
```
Comment (brentvatne) [inline docs/pages/versions/unversioned/sdk/dev-client.mdx:21]:
```suggestion
`expo-dev-client` adds various useful development tools to your debug builds:

- a configurable launcher UI, so you can launch updates (such as from [PR previews](/develop/development-builds/development-workflows#pr-previews)) and switch between development servers without needing to recompile the native app
- improved debugging tools (such as support support for [inspecting network requests](/debugging/tools/#inspecting-network-requests))
- [a powerful and extensible developer menu UI](/debugging/tools#developer-menu)

Expo documentation refers to debug builds that include `expo-dev-client` as [development builds](/develop/development-builds/introduction/).
```

**Review by brentvatne — APPROVED** [review]
looking great! thank you!

**Review by lukmccall — APPROVED** [review]

File: packages/expo-dev-menu/src/DevMenu.ts [hunk 3]
```diff
@@ -34,7 +43,11 @@ registerCallbackListener();
 
 let handlers = new Map<string, () => void>();
 
-export async function registerDevMenuItems(items: ExpoDevMenuItem[]) {
+/**
+ * A method allowing to specify custom entries in the development client menu.
```
Comment (amandeepmittal) [inline packages/expo-dev-menu/src/DevMenu.ts:47]:
```suggestion
 * A method that allows to specify custom entries in the development client menu.
```

File: packages/expo-dev-launcher/src/DevLauncher.ts [hunk 4]
```diff
@@ -2,21 +2,32 @@ import { ExpoUpdatesManifest } from 'expo-manifests';
...
+ * @hidden
+ */
 export function registerErrorHandlers() {
   console.warn(
     'DevLauncher.registerErrorHandlers has been deprecated. To enable error handlers you need to import "expo-dev-launcher" at the top of your index.js.'
   );
 }
 
+/**
+ * A method returning a boolean indicating if the current application is a development build.
```
Comment (amandeepmittal) [inline packages/expo-dev-launcher/src/DevLauncher.ts:22]:
```suggestion
 * A method that returns a boolean to indicate if the current application is a development build.
```

Let's update the verbiage to keep the language easier to read.

**Review by amandeepmittal — APPROVED** [review]
Lgtm 👍 Thank you!

Left a couple of small suggestions.

**Comment by Simek** [comment]
SDK check fail is not related, made a PR fixing it in:
* https://github.com/expo/expo/pull/26719

**Pull request merged**